tensorflow/python/data/experimental/ops/get_single_element.py

```diff
@@ -20,31 +20,94 @@
 from tensorflow.python.data.ops import dataset_ops
 from tensorflow.python.data.util import structure
 from tensorflow.python.ops import gen_dataset_ops
+from tensorflow.python.util import deprecation
 from tensorflow.python.util.tf_export import tf_export
 
 
-@tf_export("data.experimental.get_single_element")
+@tf_export("data.experimental.get_single_element", "data.get_single_element")
+@deprecation.deprecated_endpoints("data.experimental.get_single_element")
 def get_single_element(dataset):
-  """Returns the single element in `dataset` as a nested structure of tensors.
+  """Returns the single element of the `dataset` as a nested structure of tensors.
 
-  This function enables you to use a `tf.data.Dataset` in a stateless
+  The function enables you to use a `tf.data.Dataset` in a stateless
   "tensor-in tensor-out" expression, without creating an iterator.
-  This can be useful when your preprocessing transformations are expressed
-  as a `Dataset`, and you want to use the transformation at serving time.
+  This facilitates the ease of data transformation on tensors using the
+  optimized `tf.data.Dataset` abstraction on top of them.
 
-  For example:
+  For example, lets consider a `preprocessing_fn` which would take as an
+  input the raw features and returns the processed feature along with
+  it's label.
 
   ```python
-  def preprocessing_fn(input_str):
-    # ...
-    return image, label
+  def preprocessing_fn(raw_feature):
+    # ... the raw_feature is preprocessed as per the use-case
+    return feature
 
-  input_batch = ...  # input batch of BATCH_SIZE elements
-  dataset = (tf.data.Dataset.from_tensor_slices(input_batch)
+  raw_features = ...  # input batch of BATCH_SIZE elements.
+  dataset = (tf.data.Dataset.from_tensor_slices(raw_features)
              .map(preprocessing_fn, num_parallel_calls=BATCH_SIZE)
              .batch(BATCH_SIZE))
 
-  image_batch, label_batch = tf.data.experimental.get_single_element(dataset)
+  processed_features = tf.data.get_single_element(dataset)
+  ```
+
+  In the above example, the `raw_features` tensor of length=BATCH_SIZE
+  were converted to a `tf.data.Dataset`. Next, all the raw_features were
+  mapped using the `preprocessing_fn` and the processed features were
+  grouped into a single batch. The final `dataset` contains only one element
+  which is a batch of all the processed features.
+
+  NOTE: The `dataset` should contain only one element. Preferrably, a batch
```

Comment:
Remove the sentence starting with "Preferably, ..."

Reply:
addressed in the commit.